src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -547,42 +525,93 @@ private static String toLowerCase(String s) {
         return new String(chars);
     }
 
-    /**
-     * Optionally decodes a String, if it's encoded. If String is not encoded,
-     * method will return the original input value.
-     *
-     * @param input the value String to decode
-     * @return a decoded String
-     */
-    private String percentDecode(final String input) {
-        if (input == null) {
-            return null;
-        }
-        final String decoded = uriDecode(input);
-        if (!decoded.equals(input)) {
-            return decoded;
+    private static String percentDecode(final String source) {
+        if (source == null || source.isEmpty()) {
+            return source;
         }
-        return input;
-    }
 
-    public static String uriDecode(String source) {
-        if (source == null) {
+        byte[] bytes = source.getBytes(StandardCharsets.UTF_8);
```

Comment:
_Nit_: should `ByteBuffer` be also used here?

Reply:
Maybe if there's some advantage to using `ByteBuffer` over `byte[]`.

 I think that `String::getBytes(byte[])` and `Charset::encode(ByteBuffer)` actually do the same internally.